test/check_result.jl

```diff
@@ -83,6 +83,20 @@ end
                 Tangent{Tuple{Float64,Float64}}(1.0, 2.0),
                 @thunk(Tangent{Tuple{Float64,Float64}}(1.0, 2.0)),
             )
+            test_approx(
+                @thunk(Tangent{Tuple{Float64,Float64}}(1.0, 2.0)),
+                Tangent{Tuple{Float64,Float64}}(1.0, 2.0),
+            )
+            test_approx(@thunk(ZeroTangent()), ZeroTangent())
+            test_approx(ZeroTangent(), @thunk(ZeroTangent()))
+            test_approx(
+                Tangent{Tuple{Float64,Float64}}(ZeroTangent(), NoTangent()),
+                NoTangent(),
+            )
+            test_approx(
+                NoTangent(),
+                Tangent{Tuple{Float64,Float64}}(ZeroTangent(), NoTangent()),
+            )
```

Comment:
Do we want this? I think we do

Reply:
In practice I think what matters is that `NoTangent()` is the same as `Tangent` of just `NoTangent()`s.

E.g. the choice similar to the one we made in
```julia
julia> rand_tangent((1, 2))
NoTangent()
```

I added on the `ZeroTangent` on purely practical grounds, since they are effectively the same. But I can't justify it either, so we can push that decision down the road when we deal with automating testing a bit better, and restrict to `NoTangent` for now.

Does that sound good to merge?

Reply:
That sounds good